### PR TITLE
refactor multi test set datasets; add seq id test splits to GO

### PR DESCRIPTION
**Pull request by a-r-j**
* Improves API for multi test set datasets
* Improves `setup` in base datamodule to avoid setting up all datasets at once
* Adds caching to avoid duplicating in memory loading
* Adds seq id splits to GO dataset.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -557,3 +557,18 @@ To build a local version of the project's Sphinx documentation web pages:
...
+
+## Citing `ProteinWorkshop`
+
+Please consider citing `proteinworkshop` if it proves useful in your work.
+
+```bibtex
+@inproceedings{
+  jamasb2024evaluating,
+  title={Evaluating Representation Learning on the Protein Structure Universe},
+  author={Arian R. Jamasb, Alex Morehead, Zuobai Zhang, Chaitanya K. Joshi,Kieran Didi, Simon V. Mathis, Charles Harris, Jian Tang, Jianlin Cheng,Pietro Lio, Tom L. Blundell},
```
Comment (amorehead) [inline README.md:569]: Comma spacing between authors seems off?

File: proteinworkshop/datasets/base.py [hunk 2]
```diff
@@ -518,7 +530,7 @@ def get(self, idx: int) -> Data:
         :return: PyTorch Geometric Data object.
         """
         if self.in_memory:
-            return self._batch_format(self.data[idx])
+            return self._batch_format(copy.deepcopy(self.data[idx]))
```
Comment (amorehead) [inline proteinworkshop/datasets/base.py:533]: Does this lead to one of the performance enhancements we discussed previously?

File: README.md (hunk 1 above)
Comment (a-r-j) [reply README.md:569]: Eagle-eyed!

File: proteinworkshop/datasets/base.py (hunk 2 above)
Comment (a-r-j) [reply proteinworkshop/datasets/base.py:533]: No noticeable delta but it makes me feel better about some of the gotchas here: https://alecstashevsky.com/post/on-the-fly-augmentation-with-pytorch-geometric-and-lightning-what-tutorials-dont-teach/

Comment (amorehead) [reply proteinworkshop/datasets/base.py:533]: Nice. SGTM

**Pull request merged**